CMakeLists.txt

```diff
@@ -34,34 +71,36 @@ endif()
 # NOTE TO PACKAGERS: The embedded git commit hash is critical for rapid bug triage when the builds
 # can come from a variety of sources. If you are mirroring the sources or otherwise build when
 # the .git directory is not present, please comment the following line:
-include(GetGitCommitHash)
+#include(GetGitCommitHash)
 # and instead uncomment the following, adding the complete git hash of the checkout you are using:
 # set(GIT_COMMIT_HASH 0000000000000000000000000000000000000000)
 
 string(SUBSTRING "${GIT_COMMIT_HASH}" 0 8 solvespace_GIT_HASH)
-project(solvespace
+project(solvespaceQt
     VERSION 3.1
     LANGUAGES C CXX ASM)
 
+set(USE_QT_GUI        OFF CACHE BOOL
```

Comment:
The [CMakeLists.txt](https://github.com/solvespace/solvespace/pull/1451/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a) is not acceptable in this state. As it is all [checks](https://github.com/solvespace/solvespace/pull/1451/checks) fail.

The QT version needs to be an optional _non_ default target on any of the three platforms supported by QT.

Unless this option is set SolveSpace should continue to build just like before on all four platforms using the native/GTK GUI (Windows, Linux, macOS, Web/Emscripten).

Reply:
I don't know CMake well (all that is from shingen75), so it may take some time to cleanup things.

I'm currently building solvespace by other means as I was unable to get a debug build using cmake (hence https://github.com/solvespace/libdxfrw/pull/14).  I tried this...

    cmake -DOPENGL=3 -DUSE_QT_GUI=ON -DCMAKE_BUILD_TYPE=Debug ..

...but I don't get any symbols in the debugger.

Reply:
@WickedSmoke To get symbols in the debugger, just making the following changes:

diff --git a/cmake/c_flag_overrides.cmake b/cmake/c_flag_overrides.cmake
index 978b6b65..014d7ada 100644
--- a/cmake/c_flag_overrides.cmake
+++ b/cmake/c_flag_overrides.cmake
@@ -8,3 +8,5 @@ endif()
 if(EMSCRIPTEN)
     set(CMAKE_C_FLAGS_DEBUG_INIT "-g4")
 endif()
+
+set(CMAKE_C_FLAGS_DEBUG_INIT "-g")
diff --git a/cmake/cxx_flag_overrides.cmake b/cmake/cxx_flag_overrides.cmake
index 9c8d15fe..ec43ae18 100644
--- a/cmake/cxx_flag_overrides.cmake
+++ b/cmake/cxx_flag_overrides.cmake
@@ -8,3 +8,5 @@ endif()
 if(EMSCRIPTEN)
     set(CMAKE_CXX_FLAGS_DEBUG_INIT "-g4")
 endif()
+
+set(CMAKE_CXX_FLAGS_DEBUG_INIT "-g")


It's a bit of a hack, and make sure not to commit those changes!